### DISCUSSION TITLE
Will "use client" page be built as SSG?

### DISCUSSION THREAD

**Discussion by ricemountainer** [vercel/next.js#65442] [App Router]
Hi Experts,

Let me ask a question. I'm trying to implement an SSG page on App Router. The page needs a React library that seems to use `createContext` in the library. I am obliged to add `"use client"` on the `page.tsx` to build because the error `v.createContext  is not a function` happens without `"use client"`. I have doubt about this case... Will the page be built as SSG?
I just want to implement an SSG page that is built by reading static files (a JSON file) stored in the server, which is located on other directories under `src`. In my understanding, App Router doesn't support `getStaticProps` function, so I'm not sure how to implement SSG page like above (which has a React library using `createContext`) on App Router. I don't want the page to be a page which reads the JSON file every time when rendering. Is there a sample about this?

**Comment by icyJoseph** [comment]
Yes, client components are also SSR'd, theya re the components we have used until now. The new thing are server components, which do not ship their JS to the client.

- https://github.com/reactwg/server-components/discussions/4
